servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -6726,6 +6726,46 @@ public void putTenancy(ResourceContext ctx, String tenantDomain, String provider
         if (authzServiceTokenOperation) {
             setupTenantAdminPolicyInProvider(ctx, provSvcDomain, provSvcName, tenantDomain,
                     false, auditRef, caller);
+
+            // If listed as service provider, register dependency between the provider service and the tenant domain (unless dependency exist
+            // following creation of a previous resource group)
+
+            tenancyRegisterDomainDependency(ctx, tenantDomain, provider, auditRef, caller);
+        }
+    }
+
+    private void tenancyRegisterDomainDependency(ResourceContext ctx, String tenantDomain, String provider, String auditRef, String caller) {
+        try {
+            if (serviceProviderManager.isServiceProvider(provider)) {
+
+                DomainList domainList = dbService.listDomainDependencies(provider);
+                if (!domainList.getNames().contains(tenantDomain)) {
+                    dbService.putDomainDependency(ctx, tenantDomain, provider, auditRef, caller);
+                }
+            }
+        } catch (ResourceException ex) {
+            if (ex.getCode() == ResourceException.UNAUTHORIZED) {
+                LOG.warn("Not registering domain dependency: " + ex.getMessage());
+                return;
+            }
+            throw ex;
+        }
+    }
+
+    private void tenancyDeregisterDomainDependency(ResourceContext ctx, String tenantDomain, String provSvcDomain, String provSvcName, String auditRef, String caller) {
+        String serviceToDeregister = provSvcDomain + "." + provSvcName;
+        try {
+            if (serviceProviderManager.isServiceProvider(serviceToDeregister)) {
+                boolean tenantDomainRolesExist = isTenantDomainRolesExist(tenantDomain, provSvcDomain, provSvcName);
+                if (!tenantDomainRolesExist) {
+                    dbService.deleteDomainDependency(ctx, tenantDomain, serviceToDeregister, auditRef, caller);
+                }
+            }
+        } catch (ResourceException ex) {
```

Comment:
do we still need this try/catch block? I don't see which api is returning the unauthorized exception.